htpasswd.go

```diff
@@ -177,30 +216,66 @@ func (h *HtpasswdFile) ReloadCheck() error {
 func (h *HtpasswdFile) Validate(user string, password string) bool {
 	_ = h.ReloadCheck()
 
+	hash := sha256.New()
```

Comment:
This is re-inventing https://golang.org/pkg/sync/#Map

Reply:
I don't understand what the hash calculation has to do with sync.Map. The idea here is to not permanently keep the plaintext `user:password` in memory.